### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by TimmyChan99**
- Use jest framework for testing.
- Create a test file for a file containing the following functions:
     - a function for editing the task description.
     - a function for updating an item's 'completed' status.
     - the "Clear all completed" function.
- Make sure we group tests using the `description()` method.
- The tests are in the testing.test.js file.

### CODE REVIEW

File: src/modules/edit.js [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
+//     }
+//   }
+// }
+
+// export function clearAll(list) {
+//   const UncompeledTasks = list.filter((task) => task.completed === false);
+//   UncompeledTasks.forEach((task, i) => { (task.index = i + 1); });
+//   list = UncompeledTasks;
+//   return list;
+// }
```
Comment (mikethreels) [inline src/modules/edit.js]: - [ ] Please do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

File: src/index.js [hunk 2]
```diff
@@ -142,11 +142,12 @@ listTasks.addEventListener('change', (e) => {
...
 
 clearBtn.addEventListener('click', () => {
-  const UncompeledTasks = list.filter((task) => task.completed === false);
-  UncompeledTasks.forEach((task, i) => { (task.index = i + 1); });
-  list = UncompeledTasks;
-  displayTasks(list);
-  UpdateStorage(list);
+  // const UncompeledTasks = list.filter((task) => task.completed === false);
+  // UncompeledTasks.forEach((task, i) => { (task.index = i + 1); });
+  // list = UncompeledTasks;
```
Comment (mikethreels) [inline src/index.js:147]: - [ ] Please do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

File: src/index.js [hunk 3]
```diff
@@ -3,11 +3,11 @@ import more from './images/bin.png';
 // import add from './modules/add.js';
 // import removeTask from './modules/remove.js';
```
Comment (mikethreels) [inline src/index.js:4]: - [ ] Please do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

File: src/addRemove.test.js [hunk 4]
```diff
@@ -27,3 +31,93 @@ describe('testing delete', () => {
...
+//     const div = document.querySelector('.div');
+//     const input = document.querySelector('.edit');
+//     input.addEventListener('click', () => {
+//       saveUpdatedTask('task', div, input);
+//     });
+//     const event = new Event('click');
+//     input.dispatchEvent(event);
+//     expect(document.querySelector('.edit')).toBeFalsy();
+//   });
+// });
```
Comment (mikethreels) [inline src/addRemove.test.js:123]: - [ ] Please do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

File: src/addRemove.js [hunk 5]
```diff
@@ -6,4 +6,47 @@ function removeTask(arr, index) {
...
+//     }
+//   }
+// }
+
+// export function clearAll(list) {
+//   const UncompeledTasks = list.filter((task) => task.completed === false);
+//   UncompeledTasks.forEach((task, i) => { (task.index = i + 1); });
+//   list = UncompeledTasks;
+//   return list;
+// }
```
Comment (mikethreels) [inline src/addRemove.js:52]: - [ ] Please do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

File: src/testing.test.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    const div = document.querySelector('.div');
+    const input = document.querySelector('.edit');
+    input.addEventListener('click', () => {
+      saveUpdatedTask('task', div, input);
+    });
+    const event = new Event('click');
+    input.dispatchEvent(event);
+    expect(document.querySelector('.edit')).toBeFalsy();
+  });
```
Comment (mikethreels) [inline src/testing.test.js:96]: - [ ] In this testing file you are not actually testing the functions used in your project but just the functions in the testing.js file please make sure you are testing the actual functions you are using in the project.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by leolpaz — APPROVED** [review]
Hi @TimmyChan99 , 

Your project is complete!
Congratulations! 🎉🎉 

To highlight:

<Highlight good things>

Now there is nothing left to do but to merge it! 🏆 





Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**